pkg/ingester/activeseries/active_series.go

```diff
@@ -41,16 +41,22 @@ type seriesStripe struct {
 	// without holding the lock -- hence the atomic).
 	oldestEntryTs atomic.Int64
 
-	mu             sync.RWMutex
-	refs           map[uint64]seriesEntry
-	active         int   // Number of active entries in this stripe. Only decreased during purge or clear.
-	activeMatching []int // Number of active entries in this stripe matching each matcher of the configured Matchers.
+	mu                                   sync.RWMutex
+	refs                                 map[uint64]seriesEntry
+	active                               int   // Number of active entries in this stripe. Only decreased during purge or clear.
+	activeMatching                       []int // Number of active entries in this stripe matching each matcher of the configured Matchers.
+	activeNativeHistograms               int   // Number of active entries (only native histograms) in this stripe. Only decreased during purge or clear.
+	activeMatchingNativeHistograms       []int // Number of active entries (only native histograms) in this stripe matching each matcher of the configured Matchers.
+	activeNativeHistogramBuckets         int   // Number of buckets in active native histogram entries in this stripe. Only decreased during purge or clear.
+	activeMatchingNativeHistogramBuckets []int // Number of buckets in active native histogram entries in this stripe matching each matcher of the configured Matchers.
 }
 
 // seriesEntry holds a timestamp for single series.
 type seriesEntry struct {
-	nanos   *atomic.Int64        // Unix timestamp in nanoseconds. Needs to be a pointer because we don't store pointers to entries in the stripe.
-	matches preAllocDynamicSlice //  Index of the matcher matching
+	nanos                     *atomic.Int64        // Unix timestamp in nanoseconds. Needs to be a pointer because we don't store pointers to entries in the stripe.
+	matches                   preAllocDynamicSlice //  Index of the matcher matching
+	isNativeHistogram         bool
```

Comment:
On second thought @charleskorn is right about encoding this bool in the count, in particular because we have an entry per series, so that's millions of bytes of memory wasted on this.
Let's say numNativeHistogramBuckets < 0 (i.e. -1) means not a native histogram, wdyt?